File: lib/api/owly.js
var _ = require('lodash'),
  Promise = require('bluebird'),
  util = require('../util'),
  log = util.logger();

function Owly(hootsuite, connection) {
  this._hootsuite = hootsuite;
  this._connection = connection;
}

Owly.prototype = {
  shortenUrl: function (longUrl) {
    var path = util.createOwlyPath('url', 'shorten');
    return this._connection.get(path, { longUrl, _method: 'GET' });
  },

  expandUrl: function (shortUrl) {
    var path = util.createPath('url', 'expand');
    return this._connection.get(path, { shortUrl, _method: 'GET' });
  },

  getInfo: function (shortUrl) {
    var path = util.createPath('url', 'info');
    return this._connection.get(path, { shortUrl, _method: 'GET' });
  },

  getClickStats: function (shortUrl, from, to) {
    var path = util.createPath('url', 'clickStats');
    return this._connection.get(path, { shortUrl, from, to, _method: 'GET' });
  },

  uploadPhoto: function (fileName, uploaded_file) {
    var path = util.createPath('photo', 'upload');
    return this._connection.post(path, { fileName, uploaded_file, _method: 'POST' });
  },

  uploadDoc: function (fileName, uploaded_file) {
    var path = util.createPath('doc', 'upload');
    return this._connection.post(path, { fileName, uploaded_file, _method: 'POST' });
  },
}

module.exports = Owly;

File: src.net/Hootsuite.Rest/Api/Owly.cs
using Newtonsoft.Json.Linq;
using System;
using System.IO;
using System.Threading.Tasks;

namespace Hootsuite.Rest.Api
{
    public class Owly
    {
        Hootsuite _hootsuite;
        ConnectionOwly _connection;

        public Owly(Hootsuite hootsuite, ConnectionOwly connection)
        {
            _hootsuite = hootsuite;
            _connection = connection;
        }

        public Task<JObject> shortenUrl(string longUrl)
        {
            var path = util.createOwlyPath("url", "shorten");
            return _connection.get(path, new { longUrl });
        }

        public Task<JObject> expandUrl(string shortUrl)
        {
            var path = util.createPath("url", "expand");
            return _connection.get(path, new { shortUrl });
        }

        public Task<JObject> getInfo(string shortUrl)
        {
            var path = util.createPath("url", "info");
            return _connection.get(path, new { shortUrl });
        }

        public Task<JObject> getClickStats(string shortUrl, DateTime? from, DateTime? to)
        {
            var path = util.createPath("url", "clickStats");
            return _connection.get(path, new { shortUrl, from, to });
        }

        public Task<JObject> uploadPhoto(string fileName, Stream uploaded_file)
        {
            var path = util.createPath("photo", "upload");
            return _connection.post(path, new { fileName, uploaded_file });
        }

        public Task<JObject> uploadDoc(string fileName, Stream uploaded_file)
        {
            var path = util.createPath("doc", "upload");
            return _connection.post(path, new { fileName, uploaded_file });
        }
    }
}
